src/api/composition/CompositionProvider.js

```diff
@@ -19,102 +19,95 @@
  * this source code distribution or the Licensing information page available
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
+import _ from 'lodash';
+import objectUtils from "../objects/object-utils";
 
-define([
-    'lodash',
-    'objectUtils'
-], function (
-    _,
-    objectUtils
-) {
+/**
+ * @typedef {import('../objects/ObjectAPI').DomainObject} DomainObject
+ */
+
+/**
+ * @typedef {import('../objects/ObjectAPI').Identifier} Identifier
+ */
+
+/**
+ * @typedef {import('./CompositionAPI').default} CompositionAPI
+ */
+
+/**
+ * @typedef {import('../../../openmct').OpenMCT} OpenMCT
+ */
+
+/**
+ * A CompositionProvider provides the underlying implementation of
+ * composition-related behavior for certain types of domain object.
+ *
+ * By default, a composition provider will not support composition
+ * modification.  You can add support for mutation of composition by
+ * defining `add` and/or `remove` methods.
+ *
+ * If the composition of an object can change over time-- perhaps via
+ * server updates or mutation via the add/remove methods, then one must
+ * trigger events as necessary.
+ *
+ */
+export default class CompositionProvider {
     /**
-     * A CompositionProvider provides the underlying implementation of
-     * composition-related behavior for certain types of domain object.
-     *
-     * By default, a composition provider will not support composition
-     * modification.  You can add support for mutation of composition by
-     * defining `add` and/or `remove` methods.
-     *
-     * If the composition of an object can change over time-- perhaps via
-     * server updates or mutation via the add/remove methods, then one must
-     * trigger events as necessary.
-     *
-     * @interface CompositionProvider
-     * @memberof module:openmct
+     * @param {OpenMCT} publicAPI
+     * @param {CompositionAPI} compositionAPI
      */
-
-    function DefaultCompositionProvider(publicAPI, compositionAPI) {
+    constructor(publicAPI, compositionAPI) {
+        /** @type {OpenMCT} */
         this.publicAPI = publicAPI;
+        /** @type {Object.<string, DomainObject>} */
         this.listeningTo = {};
-        this.onMutation = this.onMutation.bind(this);
+        this.onMutation = this.#onMutation.bind(this);
 
-        this.cannotContainItself = this.cannotContainItself.bind(this);
-        this.supportsComposition = this.supportsComposition.bind(this);
+        this.cannotContainItself = this.#cannotContainItself.bind(this);
```

Comment:
Assuming these can't actually be private because they are referenced externally in a callback or something?

Reply:
Oop, these can actually just be private.